dbms/src/Core/Block.cpp

```diff
@@ -543,8 +543,12 @@ Block mergeBlocks(Blocks && blocks)
 
     for (size_t i = 1; i < blocks.size(); ++i)
     {
-        if (likely(blocks[i].rows()) > 0)
+        if (likely(blocks[i].rows() > 0))
         {
+            if (unlikely(!blocksHaveEqualStructure(first_block, blocks[i])))
+            {
+                throw Exception("Cannot merge different structure blocks.", ErrorCodes::LOGICAL_ERROR);
+            }
```

Comment:
Maybe we just need to check under debug mode?
```suggestion
            assert(blocksHaveEqualStructure(first_block, blocks[i]));
```

Reply:
Or use RUNTIME_CHECK
```suggestion
            RUNTIME_CHECK(blocksHaveEqualStructure(first_block, blocks[i]), "Cannot merge different structure blocks.");
```

Reply:
> Or use RUNTIME_CHECK

OK.

Reply:
> Or use RUNTIME_CHECK

done.

Reply:
> Maybe we just need to check under debug mode?

done.